wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/devfile/server/DevfileModule.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.workspace.devfile.server;
+
+import static com.google.inject.multibindings.MapBinder.newMapBinder;
+import static com.google.inject.multibindings.Multibinder.newSetBinder;
+import static org.eclipse.che.api.workspace.devfile.server.Constants.EDITOR_COMPONENT_TYPE;
+import static org.eclipse.che.api.workspace.devfile.server.Constants.PLUGIN_COMPONENT_TYPE;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.multibindings.MapBinder;
+import com.google.inject.multibindings.Multibinder;
+import org.eclipse.che.api.workspace.devfile.server.convert.DevfileConverter;
+import org.eclipse.che.api.workspace.devfile.server.convert.component.ComponentProvisioner;
+import org.eclipse.che.api.workspace.devfile.server.convert.component.ComponentToWorkspaceApplier;
+import org.eclipse.che.api.workspace.devfile.server.convert.component.editor.EditorComponentProvisioner;
+import org.eclipse.che.api.workspace.devfile.server.convert.component.editor.EditorComponentToWorkspaceApplier;
+import org.eclipse.che.api.workspace.devfile.server.convert.component.plugin.PluginComponentToWorkspaceApplier;
+import org.eclipse.che.api.workspace.devfile.server.convert.component.plugin.PluginProvisioner;
+import org.eclipse.che.api.workspace.devfile.server.validator.ComponentIntegrityValidator.NoopComponentIntegrityValidator;
+import org.eclipse.che.api.workspace.devfile.server.validator.DevfileSchemaValidator;
+import org.eclipse.che.api.workspace.server.DevfileToWorkspaceConfigConverter;
+
+/** @author Sergii Leshchenko */
+public class DevfileModule extends AbstractModule {
+
+  @Override
+  protected void configure() {
+    bind(DevfileSchemaValidator.class);
+    bind(DevfileService.class);
+
+    Multibinder<ComponentProvisioner> workspaceToDevfileAppliers =
```

Comment:
I'm not sure that we really need `DevfileBindings` but if we have it, why do not reuse it for `ComponentProvisioner` and `ComponentToWorkspaceApplier` in this class?

Reply:
Yeah, I first added `KubernetesDevfileBindings` that contains a non-trivial setup of the binding which I thought warranted making sure that we don't mess it up on the places it is needed (https://github.com/eclipse/che/blob/8f88534bc253087f19f0085686ad5148ed1dc481/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/devfile/KubernetesDevfileBindings.java#L78).

I liked the way how it read in the module definitions so I added also the trivial bindings in the `DevfileBindings`.

If you believe we should be explicit about the binding setup in the modules, I can remove these classes but I believe it makes the binding definitions ever so slightly less error prone.

Reply:
I'm OK with leaving `DevfileBindings` class